testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/saml/IdpInitiatedLoginTest.java

```diff
@@ -71,6 +69,70 @@ public void testIdpInitiatedLogin() {
         ;
     }
 
+    @Test
+    public void testIdpInitiatedLoginPostAdminUrl() {
+        ClientRepresentation clientRep = adminClient.realm(REALM_NAME).clients().findByClientId(SAML_CLIENT_ID_SALES_POST).get(0);
+        String url = clientRep.getAttributes().get(SamlProtocol.SAML_ASSERTION_CONSUMER_URL_POST_ATTRIBUTE);
+        adminClient.realm(REALM_NAME).clients().get(clientRep.getId())
+                .update(ClientBuilder.edit(clientRep)
+                        .adminUrl(url)
+                        .attribute(SamlProtocol.SAML_ASSERTION_CONSUMER_URL_POST_ATTRIBUTE, null)
+                        .attribute(SamlProtocol.SAML_ASSERTION_CONSUMER_URL_REDIRECT_ATTRIBUTE, null)
+                        .build());
+        try {
```

Comment:
replace the lines 74-82 with
```java
        try (Closeable c = ClientAttributeUpdater.forClient(adminClient, REALM_NAME, SAML_CLIENT_ID_SALES_POST)
          .setAdminUrl(url)
          .setAttribute(SamlConfigAttributes.SAML_ASSERTION_CONSUMER_URL_POST_ATTRIBUTE, null)
          .setAttribute(SamlConfigAttributes.SAML_ASSERTION_CONSUMER_URL_REDIRECT_ATTRIBUTE, null)
          .update()) {

```
(add `ClientAttributeUpdater.setAdminUrl` method similarly to other ones in `ClientAttributeUpdater` class)
Then you can remove the `finally` part.